pages/api/applications/generate-report.tsx

```diff
@@ -0,0 +1,174 @@
+  'Email',
+  'NINumber',
+  'Sex',
+  'RegistrationDate',
+  'EffectiveDate',
+  'ApplicantType',
+  'MinimumBedSize',
+  'MaximumBedSize',
+  'DateOfBirth',
+  'OlderPersonsAssessement',
+  'MobilityAssessment',
+  'AdditionalBandingInfo',
+  'MedicalRequirements',
+  'Offered',
+  'EthnicOrigin',
+  'Decant',
+  'AHRCode',
+  'AutoBidPref_MobilityStandard',
+  'AutoBidPref_WheelChairStandard',
+  'AutoBidPref_AdaptedStandard',
+] as const;
+
+type CSVRow = { [Key in typeof CSVColumns[number]]: string | null };
+
+const endpoint: NextApiHandler = async (
+  req: NextApiRequest,
+  res: NextApiResponse
+) => {
+  switch (req.method) {
+    case 'GET':
+      const user = getSession(req);
+
+      const auth = getAuth(process.env.AUTHORISED_ADMIN_GROUP as string, user);
+      if (!('user' in auth)) {
+        res.status(StatusCodes.FORBIDDEN).json({ message: 'access denied' });
+        return;
+      }
+
+      const csv = createObjectCsvStringifier({
+        header: CSVColumns.map((id) => ({ id, title: id })),
+      });
+
+      try {
+        const firstPage = await getApplications(0);
+        if (!firstPage) {
+          throw new Error('Cannot get application data');
+        }
+        let pages: CSVRow[][] = [batchToCSV(firstPage)];
+        for (let i = 1; i < firstPage.totalNumberOfPages; i++) {
+          const r = await getApplications(i);
+          if (r) {
+            pages.push(batchToCSV(r));
+          }
+        }
+
+        res.status(StatusCodes.OK);
+        res.setHeader('Content-Type', 'text/csv');
+        res.setHeader(
+          'Content-Disposition',
+          `attachment;filename=LBH-housing-register-${new Date().toLocaleDateString()}`
+        );
+        res.send(csv.getHeaderString() + csv.stringifyRecords(pages.flat()));
+      } catch (error) {
+        res
+          .status(StatusCodes.INTERNAL_SERVER_ERROR)
+          .json({ message: 'Unable to generate report' });
+      }
+      break;
+
+    default:
+      res
+        .status(StatusCodes.BAD_REQUEST)
+        .json({ message: 'Invalid request method' });
+  }
+};
+
+export default endpoint;
+
+/*
+HousingRegisterRef                u_novalet_ref
+Title                             MainApplicant
+FirstName                         MainApplicant
+FamilyName                        MainApplicant
+Address1                          MainApplicant
+Address2                          MainApplicant
+Address3                          MainApplicant
+Address4                          MainApplicant
+Address5                          MainApplicant
+Postcode                          MainApplicant
+HomeTelephone                     MainApplicant
+WorkTelephone                     [we're not collecting]
+Email                             ''
+NINumber                          MainApplicant
+Sex                               "M" or "F"
+RegistrationDate                  ? Date application made or approved?
+EffectiveDate                     ? Effective Band Date.
```

Comment:
This will be the 'application date', which can be edited by an officer and could be backdated.